src/backend/rocksdb.rs

```diff
@@ -0,0 +1,363 @@
+use crate::backend::{helpers, BlobBackend, DatabaseBackend, KvBackend};
+use crate::interop::into_tonic_status;
+use crate::proto::{blob, kv};
+use crate::{DynStream, Location, RpcResponse, StreamingRequest};
+use async_stream::stream;
+use std::sync::atomic::{AtomicBool, Ordering};
+use tonic::{async_trait, Response, Status};
+use rocksdb::DB;
+use rand::{Rng, SeedableRng};
+
+/// A backend utilizing RocksDb.
+#[derive(Debug)]
+pub struct RocksDb {
+    location: Location,
+    initialized: AtomicBool,
+}
+
+#[cfg_attr(feature = "tracing", tracing::instrument)]
+fn generate_id() -> u64 {
+    rand::rngs::SmallRng::from_entropy().r#gen()
+}
+
+impl DatabaseBackend for RocksDb {
+    type Connection = DB;
+    type Error = rocksdb::Error;
+
+    fn at_location(location: Location) -> Result<Self, Self::Error> {
+        Ok(Self {
+            location,
+            initialized: AtomicBool::new(false),
+        })
+    }
+
+    fn location(&self) -> &Location {
+        &self.location
+    }
+
+    fn connect(&self) -> Result<Self::Connection, Self::Error> {
+        match &self.location() {
+            Location::InMemory => todo!(),
```

Comment:
```suggestion
            Location::InMemory => unimplemented!(),
```

Let's leave it at this until there's a better way to handle the inability to open a database in memory.